bindings/java/lib/src/main/java/org/iota/types/AccountHandle.java

```diff
@@ -556,6 +557,29 @@ public Transaction createAliasOutput(CreateAliasOutput options) throws WalletExc
         return CustomGson.get().fromJson(callAccountMethod(options), Transaction.class);
     }
 
+    /**
+     * Asks the faucet for funds.
+     *
+     * @throws NoFundsReceivedFromFaucetException when the faucet didn't fund the address.
+     */
+    public void requestFundsFromFaucet(RequestFundsFromFaucet options) throws WalletException, NoFundsReceivedFromFaucetException {
+        int maxAttempts = 5;
+        for(int i = 0; i < maxAttempts; i++) {
+            if(syncAccount(new SyncAccount()).getBaseCoin().getAvailable() < 1000000) {
+                callAccountMethod(options);
+                try {
+                    Thread.sleep(1000 * 25);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+            } else
+                return;
+        }
+
+        if(syncAccount(new SyncAccount()).getBaseCoin().getAvailable() < 1000000)
+            throw new NoFundsReceivedFromFaucetException();
+    }
+
```

Comment:
When requesting there is no guarantee that the faucet will process the request soon.
And the Faucet also doesn't answer with a nice message that we could return to the user.
All this is really bad for the UX. Calling only the function and returning the current HTML is bad IMO. Imo we need some improved faucet REST API.
Thats the reason why I added syncing here with some retries in case there was no funding transaction.  The idea is to make the requestFundsFromFaucet() more reliable.

The user shouldn't use the requestFundsFromFaucet() to sync his account. He shouldn't care about it.

Reply:
But you're calling the sync method internally and then can cause issues

Reply:
why can it cause issues?

Reply:
> it can happen that you sync outputs which usually don't get synced (for example exchange only wants to sync basic outputs with no further unlock conditions, with the default they would now also get other outputs when calling this command)

Reply:
I'm no passing the synch options parameter, do you think this is fine like this?

Reply:
Yes, this should be fine